pkg/redis/stream.go

```diff
@@ -25,18 +26,20 @@ func NewStream(client *Client, maxLen int64) *Stream {
 	}
 }
 
-// Publish publishs a message to given topic.
-func (s *Stream) Publish(ctx context.Context, topic string, msg interface{}) error {
-	data, err := Encode(msg)
-	if err != nil {
-		return err
+// Publish publishes a message to given topic.
+func (s *Stream) Publish(ctx context.Context, cfg publisher.Config, data []byte, extra map[string]string) error {
```

Comment:
```suggestion
func (s *Stream) Publish(ctx context.Context, cfg publisher.Config, data []byte) error {
```

Reply:
I want the interface to be reusable for various message queues, which is why I need the extra mapping. It can be used to add metadata or headers to messages.

Reply:
Don't put it in the interface, it's not generalize enough.